ld_pruning.R

```diff
@@ -1,19 +1,25 @@
 #! /usr/bin/env Rscript
 library(argparser)
+library(magrittr)
 
 # read arguments
-argp <- arg_parser("LD pruning")
-argp <- add_argument(argp, "gds_file", help="GDS file")
-argp <- add_argument(argp, "--out_file", help="output file name", default="pruned_snps.rds")
-argp <- add_argument(argp, "--sample_id", help="RDS file with vector of sample.id to include")
-argp <- add_argument(argp, "--variant_id", help="RDS file with vector of variant.id to include")
-argp <- add_argument(argp, "--maf", help="minimum MAF for variants to include", default=0.05)
-argp <- add_argument(argp, "--missing", help="maximum missing call rate for variants to include", default=0.05)
-argp <- add_argument(argp, "--r_threshold", help="r threshold for LD", default=sqrt(0.1))
-argp <- add_argument(argp, "--window_size", help="window size in Mb", default=10)
+argp <- arg_parser("LD pruning") %>%
+    add_argument("gds_file", help="GDS file") %>%
+    add_argument("--chromosome", help="chromosome") %>%
+    add_argument("--out_file", help="output file name", default="pruned_snps.rds") %>%
+    add_argument("--sample_id", help="RDS file with vector of sample.id to include") %>%
+    add_argument("--variant_id", help="RDS file with vector of variant.id to include") %>%
+    add_argument("--maf", help="minimum MAF for variants to include", default=0.05) %>%
+    add_argument("--missing", help="maximum missing call rate for variants to include", default=0.05) %>%
+    add_argument("--r_threshold", help="r threshold for LD", default=sqrt(0.1)) %>%
+    add_argument("--window_size", help="window size in Mb", default=10) %>%
+    add_argument("--exclude_pca_corr_from_build", help="exclude variants in regions with high correlation with PCs (HLA, LCT, inversions), using positions from this genome build (hg18, hg19, or hg38)")
```

Comment:
This argument could also be split into two, with an "exclude_pca_corr" flag and a separate argument to select the genome build. But since the genome build is only relevant for the correlation filters, this seemed to make sense.